### PR TITLE
feat(dashboards): Add Releases top level filter

### PR DESCRIPTION
**Pull request by shruthilayaj**
Add the top level query filter for releases.
Note that selecting an option currently doesn't
do anything. Implementing some sort of pagination
on scroll will be done in a follow up.

reopening since I somehow messed up merge conflicts on this https://github.com/getsentry/sentry/pull/36319

### CODE REVIEW

**Review by narsaynorath — COMMENTED** [review]
This is looking good, just some small things to point out, but I couldn't help find a way to make the failing test pass. I left a suggestion to do a `JSON.stringify` hack to prevent the useEffect from firing a second time, but it's not a solution to why it's re-rendering with a different `selection` object

File: static/app/views/dashboardsV2/releasesSelectControl.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    t('All Releases')
+  );
+
+  return (
+    <CompactSelect
+      multiple
+      isClearable
+      isSearchable
+      isLoading={loading}
+      menuTitle={t('All Releases')}
```
[the remark this answers is not in the document]
Comment (shruthilayaj) [reply static/app/views/dashboardsV2/releasesSelectControl.tsx:30]: The teams compact select says All Teams so I might check with Dora if we should use the same to keep consistency, so leaving it as is

Screen Shot 2022-07-08 at 5 47 23 PM

**Review by narsaynorath — APPROVED** [review]
Just one comment r.e. `environment: environments` in the provider's fetch function

File: static/app/utils/releases/releasesProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+function fetchReleases(api: Client, orgSlug: string, selection: PageFilters) {
+  const {environments, projects} = selection;
+
+  return api.requestPromise(`/organizations/${orgSlug}/releases/`, {
+    method: 'GET',
+    data: {
+      sort: 'date',
+      project: projects,
+      per_page: 50,
+      environments,
```
[the remark this answers is not in the document]
Comment (shruthilayaj) [reply static/app/utils/releases/releasesProvider.tsx:20]: ah good catch

**Pull request merged**